Communications/WatchdogManager.cs

```diff
@@ -106,8 +105,7 @@ internal static void AddWatchdog(string Endpoint)
             // Check if a client attempts to add a watchdog
             if (IsClient) { throw new InvalidOperationException("Clients cannot add watchdogs"); }
             // Construct the client's watchdog packet
-            Packet WatchdogPacket = new Packet(Constants.WATCHDOG_PING, true, Endpoint);
-            WatchdogPacket.AppendData(Utilities.UtilData.ToBytes(Endpoint));
+            Packet WatchdogPacket = new PacketWriter(Constants.WATCHDOG_PING, true, Endpoint).Put(Endpoint).Packet;
```

Comment:
Since PacketWriter requires PacketReader on the other end, this will break when the other end of the connection attempts to parse the Watchdog packet, it will break as far as I can tell.

Reply:
Unlike SafePacketWriter, PacketWriter doesn’t write type ID into the packet, so the code should be equivalent to original one. I’ll double check that later.

Reply:
I verified that the new code is equivalent to original one:

```
[TestMethod]
public void TestWatchdog()
{
    string Endpoint = "helloworld";
    Packet Packet1 = new Packet(0xf0, true, Endpoint);
    Packet1.Data.AppendData(Scarlet.Utilities.UtilData.ToBytes(Endpoint));

    Packet Packet2 = new PacketWriter(0xf0, true, Endpoint).Put(Endpoint).Packet;

    byte[] Data1 = Packet1.GetForSend();
    byte[] Data2 = Packet2.GetForSend();

    Console.WriteLine(BitConverter.ToString(Data1));
    Console.WriteLine(BitConverter.ToString(Data2));

    Assert.IsTrue(Equality(Data1, Data2));
}
```

The output is:

> 5A-22-AA-D5-F0-00-68-00-65-00-6C-00-6C-00-6F-00-77-00-6F-00-72-00-6C-00-64
> 5A-22-AA-D5-F0-00-68-00-65-00-6C-00-6C-00-6F-00-77-00-6F-00-72-00-6C-00-64

Reply:
Alrighty, I didn't realize what the distinction between `SafePacketWriter` and `PacketWriter` was. I assumed they had to be defined in order to go through the `PacketScanner`, but I see now that the `PacketScanner` just takes the amount of data out of the message that the type would theoretically take.

Looks good to me.